src/components/ha-button.ts

```diff
@@ -12,6 +12,7 @@ export class HaButton extends Button {
         margin-inline-start: 0px;
         margin-inline-end: 8px;
         direction: var(--direction);
+        display: block;
```

Comment:
@bramkragten @piitaya I added this such that there is no space rendered below the icon, misaligning the icon with the text. This will fix it for all icons rendered as a slot. Does that make sense? Happy to move it to just the assist dev tool page too. 

Before:
![CleanShot 2023-07-11 at 14 43 37](https://github.com/home-assistant/frontend/assets/1444314/4ac8b671-16a0-4baf-9e62-031f853c3335)

After:
![CleanShot 2023-07-11 at 14 43 59](https://github.com/home-assistant/frontend/assets/1444314/251783fb-5fde-42df-8c09-f00354355383)